icu4c/source/i18n/number_usageprefs.cpp

```diff
@@ -94,13 +94,12 @@ void mixedMeasuresToMicros(const MaybeStackVector<Measure> &measures, DecimalQua
         U_ASSERT(micros->outputUnit.getComplexity(status) == UMEASURE_UNIT_MIXED);
         U_ASSERT(U_SUCCESS(status));
         // Check that we received measurements with the expected MeasureUnits:
-        int32_t singleUnitsCount;
-        LocalArray<MeasureUnit> singleUnits =
-            micros->outputUnit.splitToSingleUnits(singleUnitsCount, status);
+        MeasureUnitImpl temp;
+        const MeasureUnitImpl& impl = MeasureUnitImpl::forMeasureUnit(micros->outputUnit, temp, status);
```

Comment:
Hmm, what's the best resolution for this:

```
number_usageprefs.cpp:98:32: error: unused variable 'impl' [-Werror,-Wunused-variable]
        const MeasureUnitImpl& impl = MeasureUnitImpl::forMeasureUnit(micros->outputUnit, temp, status);
```

This impl is only used in U_ASSERT, which is why I placed the whole block in a U_DEBUG. However I now see U_DEBUG can be defined while U_ASSERT is skipped:

```
#if U_DEBUG
#   include <assert.h>
#   define U_ASSERT(exp) assert(exp)
#elif U_CPLUSPLUS_VERSION
#   define U_ASSERT(exp) (void)0
#else
#   define U_ASSERT(exp)
#endif
```

I must have missed a warning for `singleUnits` then?

Can one have multiple statements together in a U_ASSERT? Perhaps drop the U_ASSERT and rely only on the U_DEBUG guards?